chore(deps): update default vm generations off v3

Carus11 opened the pull request:
Fixes #404

The v3 instances are now hidden away in the Microsoft docs under previous and retired generations. The default generation produced by sizing these days are the v5's.

dhoucgitter commented:
Hi @Carus11 , expecting that this request would have us update any surviving _v3 references in our sample.*.tfvars files under the examples folder to _v5 types also?

Carus11 commented:
Good catch, updated those in the branch. 
Additionally, I updated the connect sample even though the documentation states:
> Note: The functionality associated with the optional Connect workload class has been deprecated and is no longer available in 2024.09 and later.

Thats probably worth of another task to clean it up.

PhanikumarThodupunoori commented:
Hi Carus,

Given that the _v5 SKU is not available in many regions, I suggest update the PR to use the _v4 SKU, which is available in most of the regions, to avoid deployment issues. Additionally, the current default values in variables.tf are expected to work across all regions. Setting _v5 SKU as the default conflicts with this expectation.

Please provide your feedback, and let us know if you can update the PR accordingly. Once updated, Please let us know. Thank you!

Carus11 commented:
Just running through and checking from the product availability graphs,
- Dsv5 has greater availability worldwide than Dsv4
- Esv5 has greater availability worldwide than Esv4

The v5 instances were significantly upgraded in their ability to work with storage over the network. The are a minimum generation version I suggest to customers because of this. The I/O improvements are soo great that I would prefer to leave v5 as the example generation to strive for when available. Its perfectly reasonable for users to adjust the instances to suit their specific constraints when working in one of the few regions with constrained product lists.

Carus11 commented:
>  Additionally, the current default values in variables.tf are expected to work across all regions

If this is a requirement, then it is expected to work across all regions then leaving them on v3 would be the closest method of satisfying that, as its only missing it appears in one region in china, and the one new zealand region which doesnt have many other products listed either.

Carus11 commented:
V3 is the original value I was hoping to move forward from. But due to the requirement mentioned of needing the greatest global availability as the default I will simply close this PR.

PhanikumarThodupunoori commented:
We’ve reviewed this further and concluded that proceeding with the v5 changes for the servers is the right direction to align with our objectives. However, for flexible SKUs, we will keep them as v3 for better availability.
 
We’ll make a small commit to incorporate these updates before merging.

PhanikumarThodupunoori commented:
We made the change to the flex server instance size to revert it back to v3. We've been able to create clusters using the sample-input-minimal and sample-input-connect files and they successfully use the v5 instance size.

saschjmil reviewed (approved):
LGTM. Thanks for the update Carus!
